### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by saakshigupta2002**

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme:
Excellent

Commit messages: 

- Excellent, detailed.
- No pull request message was provided. You will need to provide:"pull-request, with clear description, completed properly".

Code:
- Uses vision transformer
- Well commented
- Good model design

Functionality/Performance:
- Test ACC: 57.63% 
- Preprocssing data correctly
- No patient level split.
- Model is underfitting. Only trained for 10 epochs
- Val acc is low 
- Loss plot shows model didn't converge
- No hyperparameter tuning shown

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem, but poor performance < 0.6 -1
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage, no patient level split -1
Difficulty: Hard
## Commit Log
Meaningful commit messages sometimes -1
Progressive commits used
## Documentation
ReadMe good, but could use usage info -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required, move readme to folder and restore repo README -2
Request Description missing -2

**Comment by shakes76** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks granted.
